File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        vector<int> bfs;
        if(!root){
            return {};
        }
        queue<TreeNode*> q;
        TreeNode* temp = root;
        q.push(temp);
        while(!q.empty()){
            int level = q.size();
            for(int i=0; i<level ;i++){
                TreeNode* curr = q.front();
                q.pop();
                if(i == level -1){
                    bfs.push_back(curr->val);
                }
                if(curr->left){
                    q.push(curr->left);
                }
                if(curr->right){
                    q.push(curr->right);
                }
            
            }
            
        }
        return bfs;
    }
};

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        dummy = node = TreeNode()
        q = deque()
        if root is None:
            return []
        if root.left is None and root.right is None:
            return [root.val]

        q.append(root)
        view=[]

        while q:
            child_q =deque()
            prev = -1
            while q:
                curr = q.popleft()
                if curr.left is not None:
                    child_q.append(curr.left)
                if curr.right is not None:
                    child_q.append(curr.right)
                prev = curr
            view.append(prev.val)
            q=child_q
        return view

